tools/c7n_tencentcloud/tests/test_tc_tags.py

```diff
@@ -0,0 +1,303 @@
+                    {
+                        "InstanceId": "ins-3zljmyzw"
+                    }
+                ],
+                "actions": [
+                    {
+                        "type": "mark-for-op",
+                        "op": "terminate",
+                        "days": 7
+                    },
+                    {
+                        "type": "stop"
+                    }
+                ]
+            },
+            config=options
+        )
+        policy.run()
+
+        resources = self.cvm.resources()
+        assert resources[0]["InstanceState"] == "STOPPING" or \
+               resources[0]["InstanceState"] == "STOPPED"
+        assert resources[0]["Tags"][0]["Key"] == "maid_status"
+
+    @pytest.mark.vcr
+    def test_cvm_marked_op_terminate(self, options):
+        policy = {
+            "filters": [
+                {
+                    "InstanceId": "ins-3zljmyzw"
+                }
+            ]
+        }
+        cvm = CVM(self.ctx, policy)
+        assert len(cvm.resources()) == 1
+
+        policy = self.load_policy(
+            {
+                "name": "cvm-marked-for-op-terminate",
+                "resource": "tencentcloud.cvm",
+                "filters": [
+                    {
+                        "type": "marked-for-op",
+                        "op": "terminate",
+                        "skew": 7
+                    }
+                ],
+                "actions": [
+                    {
+                        "type": "terminate"
+                    }
+                ]
+            },
+            config=options
+        )
+        policy.run()
+
+        assert len(cvm.resources()) == 0
+
+    @pytest.mark.vcr
+    def test_add_tag(self, options):
+        resources = self.cvm.resources()
+        for tag in resources[0]["Tags"]:
+            assert tag["Key"] != "tag_add_test_key_for_test"
+
+        policy = self.load_policy(
+            {
+                "name": "cvm-test-tag",
+                "resource": "tencentcloud.cvm",
+                "filters": [
+                    {
+                        "InstanceId": "ins-00lycyy6",
+                    }
+                ],
+                "actions": [
+                    {
+                        "type": "tag",
+                        "key": "tag_add_test_key_for_test",
+                        "value": "tag_add_test_value_for_test"
+                    }
+                ]
+            },
+            config=options
+        )
+        policy.run()
+
+        resources = self.cvm.resources()
+        tag_add_success = False
+        for tag in resources[0]["Tags"]:
+            if tag["Key"] == "tag_add_test_key_for_test" and \
+                    tag["Value"] == "tag_add_test_value_for_test":
+                tag_add_success = True
+                break
+        assert tag_add_success
+
+    @pytest.mark.vcr
+    def test_modify_tag(self, options):
+        resources = self.cvm.resources()
```

Comment:
remove this and use a filter in the policy for the tag existence.

Reply:
what we want to do here is:
1. before testing start, verify the cvm initial state
2. after operation done, check if it is the expected state 

so we use _self.cvm.resources_ to get the cvm's metadata

Reply:
we have updated code

Reply:
the fetch the modified bit with a client is still something needing changes, I will try and address in my separate updates to this pr/branch.